ArjunaJTA/jms/src/main/java/org/jboss/narayana/jta/jms/JmsXAResourceRecoveryHelper.java

```diff
@@ -69,13 +80,24 @@ public boolean initialise(String properties) {
         return new XAResource[] { this };
     }
 
+    /**
+     * Delegates XAResource#recover call to the connected JMS resource. Additionally, if provided argument is
+     * XAResource.TMSTARTRSCAN, then JMS connection will be initiated for all further communications. Or if provided argument is
+     * XAResource.TMENDRSCAN, then JMS connection will be closed at the end of the call.
+     *
```

Comment:
Remark: I assume that the calls are under the control of periodic recovery and therefore no synchronization is required to protect against recover calls interleaving.